src/Sarif/NotYetAutoGenerated/SarifRewritingVisitor.cs

```diff
@@ -756,13 +756,11 @@ public virtual Run VisitRun(Run node)
 
                 if (node.LogicalLocations != null)
                 {
-                    var keys = node.LogicalLocations.Keys.ToArray();
-                    foreach (var key in keys)
+                    if (node.LogicalLocations != null)
```

Comment:
>LogicalLocations [](start = 29, length = 16)

i have to manually update this visitor because our jschema code does not have the logical to properly visit all dictionaries (where the key value might be updated). the more dictionary to array conversion work we do, the less we need this feature and may be able to drop the request altogether.

Reply:
1. I don't actually understand this comment. It seems to be saying something about how the code that JSchema generates treats dictionaries, but logicalLocations is now an array -- Oh! you are saying that if only _all_ the dictionaries were arrays, we wouldn't need this hand-coded visitor, and this is a step in the right direction. Got it.

2. No need to repeat the test from line 757.

---
In reply to: [243700311](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243700311) [](ancestors = 243700311)